src/ui/React/LinearForwardProgress.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from "react";
+import { useState, useRef, useEffect, useCallback } from "react";
+import LinearProgress from "@mui/material/LinearProgress";
+
+interface LinearForwardProgressProps {
+  value: number;
+  backgroundColor?: React.CSSProperties["color"];
+  barColor?: React.CSSProperties["color"];
+}
+
+export const LinearForwardProgress = React.forwardRef<unknown, LinearForwardProgressProps>(
+  function LinearForwardProgress({ value, backgroundColor, barColor, ...otherProps }, ref): React.ReactElement {
+    const TRANSITION_TO_FULL_DURATION = 200;
+    const ONE_FRAME_RESET_DELAY = 16;
+
+    const [displayProgress, setDisplayProgress] = useState(value);
+    const [isInDecreasingAnimation, setIsInDecreasingAnimation] = useState(false);
+    const [isAnimatingToFull, setIsAnimatingToFull] = useState(false);
+
+    const previousProgressRef = useRef(value);
+    const animationTimeouts = useRef<number[]>([]);
+
+    const clearAnimationTimeouts = useCallback(() => {
+      animationTimeouts.current.forEach((timeout) => {
```

Comment:
```suggestion
      for (const timeout of animationTimeouts.current) {
```
Optional: forEach, specifically, has several gotchas such that I recommend for..of as the superior alternative.